File: coding-practice/50 Coding Challenges part 1/C#/problem18.cs
// coding challenge 18 : print the first 100 prime numbers
using System;
class problem18
{
    static bool is_prime(int number){
        if(number<2){
            return false;
        }

        else if(number==2){
            return true;
        }

        int max_div = (int)Math.Sqrt(number);
        for(int i=2; i<=max_div; i++){
            if(number%i == 0){
                return false;
            }
        }
        return true;
    }
    static void Main(string[] args){
        int n_prime = 100;
        int[] array_prime = new int[n_prime];
        int j = 0;
        for(int i=0; i<n_prime;){
            if(is_prime(j) == true){
                array_prime[i] = j;
                i++;
            }
            j++;
        }
        Console.WriteLine("The first "+n_prime+" prime numbers are:");
        for(int i=0; i<n_prime; i++){
            Console.WriteLine(array_prime[i]);
        }

    }
}

File: coding-practice/50 Coding Challenges part 1/Python/problem18.py
# print the first 100 prime numbers
import math
def is_prime(x):
    if x < 2:
        return False
    elif x == 2:
        return True
    for i in range(2, int(math.sqrt(x) + 1)):
        if x%i == 0:
            return False
    return True

def print_prime(n_primes):
    n = 0
    i = 2
    output = []
    while(n < n_primes):
        if is_prime(i):
            output.append(i) if is_prime(i) == True else None
            n += 1
        i += 1
    return output

output = print_prime(100)
print(len(output))
print(output)
